src/gsy_e/models/market/future.py

```diff
@@ -100,25 +162,11 @@ def _remove_old_orders_from_list(order_list: List, current_market_time_slot: Dat
             order for order in order_list if order.time_slot > current_market_time_slot
         ]
 
-    @staticmethod
-    def _remove_old_orders_from_dict(order_dict: Dict, current_market_time_slot: DateTime) -> Dict:
-        return {
-            order_id: order
-            for order_id, order in order_dict.items()
-            if order.time_slot > current_market_time_slot
-        }
-
     def delete_orders_in_old_future_markets(self, current_market_time_slot: DateTime) -> None:
         """Delete order and trade buffers."""
-        self._delete_order_dict_market_slot(current_market_time_slot,
-                                            self.slot_bid_mapping, Bid)
-        self._delete_order_dict_market_slot(current_market_time_slot,
-                                            self.slot_offer_mapping, Offer)
-        self._delete_order_dict_market_slot(current_market_time_slot,
-                                            self.slot_trade_mapping, Trade)
 
-        self.offers = self._remove_old_orders_from_dict(self.offers, current_market_time_slot)
-        self.bids = self._remove_old_orders_from_dict(self.bids, current_market_time_slot)
+        self.offers.expire(current_market_time_slot)
+        self.bids.expire(current_market_time_slot)
```

Comment:
what happened with the trades?

Reply:
The `trades` in its nature (inherited from the MarketBase) is a list, not a dictionary like the case of bids and offers
I kept dealing with the trades as usual [here](https://github.com/gridsingularity/gsy-e/blob/c860f7d45957f736c97b7a1fecdd6a73e2b63844/src/gsy_e/models/market/future.py#L177)

Reply:
However, the slot_trades_mapping is now a dynamic property, so it's always in sync with the trades list (I checked the usages of this mapping and it seems to be only used when exporting the data)